File: Arrays and misc/fib.java
// 3. Fibonacci Number

// { Driver Code Starts
//Initial Template for Java

import java.util.Scanner;

class Main {
    
// } Driver Code Ends
//User function Template for Java

public static int fibonacci(int n){
    if(n == 1 || n == 2)
        return 1;
    else
        return fibonacci(n-1) + fibonacci(n-2);
}

// { Driver Code Starts.

    public static void main(String[] args)
    {
        Scanner scn = new Scanner(System.in);
        int t = scn.nextInt();
        while(t-- > 0) {
            int n = scn.nextInt();
            int ans = fibonacci(n);
            System.out.println(ans);
        }
        scn.close();
    }
}  // } Driver Code Ends

File: Recursion/fib.cpp
// Fibonacci Using Recursion

//Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
//User function Template for C++

class Solution{
    public:
    //Complete this function
    int fibonacci(int n)
    {
        if(n <= 1)
        {
            return n;    
        }
        else
        {
            return fibonacci(n-2) + fibonacci(n-1);
        }
    }
};

// { Driver Code Starts.
int main() {
	int T;
	
	//taking total testcases
	cin>>T;
	while (T--)
	{
	    int n;
	    //taking number n
	    cin>>n;
	    Solution ob;
	    //calling fibonacci() function
	    cout<<ob.fibonacci(n)<<endl;
	    
	    
	}
	return 0;
}  // } Driver Code Ends